ncat/ncat_listen.c

```diff
@@ -621,6 +627,65 @@ int read_stdin(void)
     return nbytes;
 }
 
+/* Read from stdin and broadcast to all client sockets. Return the number of
+   bytes read, or -1 on error. */
+int read_stdin_delimiter(void)
+{
+    char buf[DEFAULT_TCP_BUF_LEN];
+
+    /* Converting the ascii valued delimiter parameter to character. */
+    char delimiter = (int)o.delimiter;
```

Comment:
This casts an `int` to an `int`. I think you meant to cast to a `char`. But the function would be nicer if `delimiter` was an argument. Make the caller do the casting.